csrc/scheduler/matmul_heuristic_plugin.cpp

```diff
@@ -0,0 +1,287 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+
+#include <ir/interface_nodes.h>
+#include <mma_type.h>
+#include <scheduler/matmul_heuristic_plugin.h>
+#include <scheduler/mma_utils.h>
+#include <utils.h>
+
+#include <dlfcn.h>
+#include <cstdint>
+
+namespace nvfuser {
+
+namespace matmul_heuristic_plugin {
+
+namespace {
+
+//! Defines HeuristicFuncPtr as type of the "getConfig" symbol
+typedef void (*HeuristicFuncPtr)(KernelConfig*, const ProblemDescription*);
+
+thread_local class PluginInterface {
```

Comment:
Should this inherit this class?
https://github.com/NVIDIA/Fuser/blob/618c4b79a053692bf83ebb32aa19dcc907ebb534/csrc/utils.h#L90-L103

Reply:
Good idea. If it gets copied we could call dlclose twice, which would be fine if we also called dlopen twice since dlopen is reference counted but the copy constructor would not call dlopen. I'll make the change.